src/group/controller/collection.ts

```diff
@@ -68,19 +68,26 @@ class GroupMemberCollectionController extends Controller {
       throw new BadRequest('This endpoint only exists for groups');
     }
 
-    if (ctx.request.body.operation === undefined) {
+    const requestBody = ctx.request.body;
+
+    if (requestBody.operation === undefined) {
       throw new UnprocessableEntity('You must specify an "operation" property');
     }
-    if (ctx.request.body.operation !== 'add-member') {
+    if (requestBody.operation !== 'add-member') {
       throw new UnprocessableEntity('"operation" must be set to "add-member"');
     }
-    if (ctx.request.body.userHref === undefined) {
+    if (requestBody.userHref === undefined) {
       throw new UnprocessableEntity('The "userHref" property must be set, and must point to the member you are adding to the group.');
     }
 
-    const newMemberHref = ctx.request.body.userHref;
-    let newMember;
+    let newMemberHref;
+    if (ctx.request.body.userHref.includes(process.env.PUBLIC_URI)) {
+      newMemberHref = requestBody.userHref.split(process.env.PORT).pop();
```

Comment:
100% this is bad. There is no port in the url in most production environments. Use `new URL()` to parse the url and extract the information you need.

I also think handling urls is actually done in the services, so it's weird that the logic for this piece of this in the controller. I think it just needs to be done in `userService` (getByHref or getByUri) so it works magically everywhere.